orangecontrib/text/tests/test_pubmed.py

```diff
@@ -0,0 +1,452 @@
+        pub_date_start = '2011/07/07'
+        pub_date_end = '2014/07/07'
+        offset = 0
+        num_requested_records = 5
+
+        # Attempt to retrieve without searching first.
+        self.assertRaises(
+                ValueError,
+                self.pubmed._retrieve_record_batch,
+                offset,
+                num_requested_records
+        )
+
+        # Must search for records first.
+        self.pubmed._search_for_records(
+                test_terms,
+                authors,
+                pub_date_start,
+                pub_date_end
+        )
+
+        # Retrieve the records.
+        data = self.pubmed._retrieve_record_batch(
+                offset,
+                num_requested_records
+        )
+
+        self.assertEqual(len(data), num_requested_records)
+
+    @patch('Bio.Entrez.esearch', mock_entrez.esearch)
+    @patch('Bio.Entrez.read', mock_entrez.read)
+    @patch('Bio.Entrez.efetch', mock_entrez.efetch)
+    @patch('Bio.Medline.parse', mock_entrez.parse)
+    @patch('Bio.Entrez.epost', mock_entrez.epost)
+    def test_pubmed_retrieve_records(self):
+        test_terms = ['orchid']
+        authors = []
+        pub_date_start = '2011/07/07'
+        pub_date_end = '2014/07/07'
+
+        num_records = 5
+
+        # Must search for records first.
+        self.pubmed._search_for_records(
+                test_terms,
+                authors,
+                pub_date_start,
+                pub_date_end
+        )
+
+        # Retrieve the records and build a corpus.
+        corpus = self.pubmed._retrieve_records(num_records)
+        self.assertEqual(len(corpus), num_records)
+
+        meta_fields = sorted([field_name
+                              for field_name, field_tag
+                              in PUBMED_TEXT_FIELDS])
+        test_meta_fields = sorted([m.name
+                                   for m
+                                   in corpus.domain.metas])
+        self.assertEqual(meta_fields, test_meta_fields)
+
+    @patch('Bio.Entrez.esearch', mock_entrez.esearch)
+    @patch('Bio.Entrez.read', mock_entrez.read)
+    @patch('Bio.Entrez.efetch', mock_entrez.efetch)
+    @patch('Bio.Medline.parse', mock_entrez.parse)
+    @patch('Bio.Entrez.epost', mock_entrez.epost)
+    def test_pubmed_retrieve_records_no_cache(self):
+        test_terms = ['orchid']
+        authors = []
+        pub_date_start = '2011/07/07'
+        pub_date_end = '2014/07/07'
+
+        num_records = 5
+
+        # Must search for records first.
+        self.pubmed._search_for_records(
+                test_terms,
+                authors,
+                pub_date_start,
+                pub_date_end
+        )
+
+        # Retrieve the records and build a corpus.
+        corpus = self.pubmed._retrieve_records(
+                num_records,
+                use_cache=False
+        )
+        self.assertEqual(len(corpus), num_records)
+
+        meta_fields = sorted([field_name
```

Comment:
Why `sorted`? Shouldn't they be stored in the same order as in `PUBMED_TEXT_FIELDS`?